docs/rfcs/bun-bundler.ts

```diff
@@ -0,0 +1,235 @@
+/**
+ * Bundler API
+ *
+ * This is a proposal for the JavaScript API for Bun's native bundler.
+ */
+
+import { BunPlugin, FileBlob } from "bun";
+
+export type JavaScriptLoader = "jsx" | "js" | "ts" | "tsx";
+export type MacroMap = Record<string, Record<string, string>>;
+export type Target = "bun" | "browser" | "node" | "neutral";
+export type Format = "iife" | "cjs" | "esm";
+export type JsxTransform = "transform" | "preserve" | "automatic";
+export type Loader =
+  | "base64"
+  | "binary"
+  | "copy"
+  | "css"
+  | "dataurl"
+  | "default"
+  | "empty"
+  | "file"
+  | "js"
+  | "json"
+  | "jsx"
+  | "text"
+  | "ts"
+  | "tsx";
+export type LogLevel = "verbose" | "debug" | "info" | "warning" | "error" | "silent";
+export type Charset = "ascii" | "utf8";
+
+export interface BundlerOptions {
+  // currently this only accepts a Loader string
+  // change: set an ext->loader map
+  loader?: { [k in string]: Loader };
+
+  // rename `platform` to `target`
+  target?:
+    | Target
+    | {
+        name: Target; // e.g. 'node', 'browser'
+        // instead of esbuild's `target`
+        version?: string; // e.g. '18', 'es2020'
+        // low priority
+        supported?: Record<string, boolean>;
+      };
+
+  // currently accepts a tsconfig as stringified JSON or an object
+  // IMO this should accept a file path to a tsconfig.json file instead
+  // or at least in addition
+  // that's what most people are used to in config files
+  tsconfig?: string;
+
+  // I don't know what this is
+  allowBunRuntime?: boolean;
+
+  ////////////////////////////////
+  // no changes below this line //
+  ////////////////////////////////
+  macro?: MacroMap;
+  autoImportJSX?: boolean;
+  trimUnusedImports?: boolean;
+  jsxOptimizationInline?: boolean;
+  inline?: boolean;
+
+  sourcemap?:
+    | boolean
+    | {
+        root?: string;
+        inline?: boolean;
+        external?: boolean;
+
+        // probably unnecessary
+        content?: boolean;
+      };
+
+  format?:
+    | Format
+    | {
+        type: "iife";
+        globalName?: string;
+      };
+
+  logging?: {
+    color?: boolean;
+    level?: LogLevel;
+    // extremely low priority
+    limit?: number;
+    override?: Record<string, LogLevel>;
+  };
+
+  // removed: mangleProps, reserveProps, mangleQuoted, mangleCache
+
+  /** Documentation: https://esbuild.github.io/api/#minify */
+  minify?:
+    | boolean
+    | {
+        whitespace?: boolean;
+        identifiers?: boolean;
+        syntax?: boolean;
+      };
+
+  treeShaking?:
+    | boolean
+    | {
```

Comment:
only a boolean

dead code elimination isn't really optional. code won't work without it